designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/QuestDbFEStatisticsRepository.scala

```diff
@@ -146,34 +142,78 @@ object QuestDbFEStatisticsRepository extends LazyLogging {
        |    WHERE timestamp_floor('d', ts) = timestamp_floor('d', now())
        | GROUP BY name""".stripMargin
 
-  def create(system: ActorSystem, clock: Clock): Resource[IO, FEStatisticsRepository[Future]] = for {
-    executorService <- createExecutorService()
-    cairoEngine     <- createCairoEngine()
-    repository      <- createRepository(executorService, cairoEngine, clock)
-    // TODO: move task properties to configuration
-    _ <- scheduleTask(system, Duration(2L, TimeUnit.SECONDS), () => repository.flushDataToDisk())
-    _ <- scheduleTask(system, Duration(2L, TimeUnit.SECONDS), () => repository.scheduleRetention())
-  } yield repository
-
-  // todo move this ExecutorService properties to a configuration
-  // todo log if task rejected
-  private def createExecutorService(): Resource[IO, ExecutionContextExecutorService] = Resource.make(
-    acquire = for {
-      executorService <- IO(new ThreadPoolExecutor(2, 4, 60L, TimeUnit.SECONDS, new ArrayBlockingQueue[Runnable](8)))
-      ec = ExecutionContext.fromExecutorService(executorService)
-    } yield ec
-  )(release = ec => IO(ec.shutdown()))
-
-  private def createCairoEngine(): Resource[IO, AtomicReference[CairoEngine]] = Resource.make(
-    acquire = IO(
-      new AtomicReference(
-        buildCairoEngine(createDirAndConfigureLogging())
-      )
-    )
-  )(release = engine => IO(closeCairoEngine(engine.get())))
+  def create(system: ActorSystem, clock: Clock, config: Config): Resource[IO, FEStatisticsRepository[Future]] =
+    for {
+      questDbConfig <- Resource.eval(IO(QuestDbConfig.apply(config)))
+      repository <- questDbConfig match {
+        case enabledCfg @ QuestDbConfig.Enabled(_, _, _, _, _) =>
```

Comment:
you can also do it like that: `case enabledCfg:QuestDbConfig.Enabled =>`

Reply:
I will change it in next PR